docs/registry/create/_index.md

```diff
@@ -773,21 +1062,94 @@ Use Go to [compile](https://pkg.go.dev/cmd/go#hdr-Compile_packages_and_dependenc
 <file>main.go</file> adds the custom model <file>mybase.go</file> from the resource registry, while <file>mybase.go</file> defines and registers the module.
 Expand the [Go module code](#code-a-main-entry-point-program) to view <file>main.go</file> for an example of this.
 
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+To compile your C++ module's executable, you must create a <file>CMakeLists.txt</file> file to handle compilation configuration, a <file>run.sh</file> file to wrap your executable, and then compile the executable using the C++ SDK build steps.
+
+1. Create a <file>CMakeLists.txt</file> file in your module directory to instruct the compiler how to compile your module.
+
+   The following example shows a basic configuration that automatically downloads the C++ SDK and handles compile-time linking for a module named `my-module`:
+
+   ```cpp {class="line-numbers linkable-line-numbers"}
+   cmake_minimum_required(VERSION 3.7 FATAL_ERROR)
+
+   project(my-module LANGUAGES CXX)
+
+   set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+
+   include(FetchContent)
+   FetchContent_Declare(
+     viam-cpp-sdk
+     GIT_REPOSITORY https://github.com/viamrobotics/viam-cpp-sdk.git
+     GIT_TAG main
+     # SOURCE_DIR ${CMAKE_SOURCE_DIR}/../viam-cpp-sdk
+     CMAKE_ARGS -DVIAMCPPSDK_USE_DYNAMIC_PROTOS=ON
+     FIND_PACKAGE_ARGS
+   )
+   FetchContent_MakeAvailable(viam-cpp-sdk)
+
+   FILE(GLOB sources *.cpp)
+   add_executable(my-module ${sources})
+   target_link_libraries(my-module PRIVATE viam-cpp-sdk::viamsdk)
+   ```
+
+2. Create a <file>run.sh</file> file in your module directory to wrap the executable and perform basic sanity checks at runtime.
+
+   The following example shows a simple configuration that performs some basic sanity checks and handles system-level linking for a module named `my-module`:
+
+   ```sh { class="command-line"}
+   #!/usr/bin/env bash
+   # run.sh -- entrypoint wrapper for the module
+
+   # bash safe mode
+   set -euo pipefail
+
+   cd $(dirname $0)
+   # get bundled .so files from this directory
+   export LD_LIBRARY_PATH=${LD_LIBRARY_PATH-}:$PWD
```

Comment:
Did you find this line to be necessary for building? I don't really know what it does.

Reply:
You're right, we're not building any `.so`s for this example, I'll remove this!

Reply:
Ah did you want to remove this `export` and the comment above?